addons/payment_sips/models/payment_acquirer.py

```diff
@@ -0,0 +1,62 @@
+# coding: utf-8
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+# Copyright 2015 Eezee-It
+
+from hashlib import sha256
+
+from odoo import models, fields, api
+from odoo.tools.translate import _
+from odoo.addons.payment.models.payment_acquirer import ValidationError
+
+from .const import SIPS_SUPPORTED_CURRENCIES
+
+class AcquirerSips(models.Model):
+    _inherit = "payment.acquirer"
+
+    provider = fields.Selection(selection_add=[("sips", "Sips")],
+        ondelete={"sips": "set default"})
+    sips_merchant_id = fields.Char("Merchant ID", required_if_provider="sips",
+        groups="base.group_user")
+    sips_secret = fields.Char("Secret Key", size=64, required_if_provider="sips",
+        groups="base.group_user")
+    sips_test_url = fields.Char("Test url", required_if_provider="sips",
+        default="https://payment-webinit.simu.sips-atos.com/paymentInit")
+    sips_prod_url = fields.Char("Production url", required_if_provider="sips",
+        default="https://payment-webinit.sips-atos.com/paymentInit")
+    sips_version = fields.Char("Interface Version", required_if_provider="sips",
+        default="HP_2.31")
+    sips_key_version = fields.Integer("Secret Key Version", required_if_provider="sips",
+        default=2)
+
+    def _sips_generate_shasign(self, values):
+        """ Generate the shasign for incoming or outgoing communications.
+
+        :param dict values: transaction values
+        :return: shasign
+        :rtype: str
+        """
+        self.ensure_one()
+        if self.provider != "sips":
+            raise ValidationError(_("Incorrect payment acquirer provider"))
+        data = values["Data"]
+        key = self.sips_secret
+
+        shasign = sha256((data + key).encode("utf-8"))
+        return shasign.hexdigest()
+
+    @api.model
+    def _sips_supported_currencies(self):
```

Comment:
Nitpicking:
```suggestion
    def _sips_get_supported_currencies(self):
```
Also, the method doesn't seem to be super useful since its only purpose is to cast a namedtuple as a recordset, and it is only called once.